### PR TITLE
feat: Upload benchmark lineages from SDK to studio

### PR DESCRIPTION
**Pull request by MerlinKallenbornAA**
# Description
TASK: PHS-885
This MR introduces  a new function to upload benchmark lineages to Studio.
It allows batch upload if the request exceeds a user defined payload size and is added in the scope of the benchmark execution upload itself.

## Before Merging
 - [x] Review the code changes
    - Unused print / comments / TODOs
    - Missing docstrings for functions that should have them
    - Consistent variable names
    - ...
 - [x] Update `changelog.md` if necessary
 - [x] Commit messages should contain a semantic [label](https://gist.github.com/joshbuchea/6f47e86d2510bce28f8e7f42ae84c716) and the ticket number
   - Consider squashing if this is not the case

### CODE REVIEW

File: src/intelligence_layer/connectors/studio/studio.py [hunk 1]
```diff
@@ -464,6 +486,78 @@ def create_benchmark_execution(
         self._raise_for_status(response)
         return str(response.json())
 
+    def submit_benchmark_lineages(
+        self,
+        benchmark_lineages: Sequence[BenchmarkLineage],
+        benchmark_id: str,
+        execution_id: str,
+        max_payload_size: int = 50 * 1024 * 1024,
```
Comment (maxhammeralephalpha) [inline src/intelligence_layer/connectors/studio/studio.py:494]: add comment that we chose this size due to studio 50mb limit

File: src/intelligence_layer/evaluation/benchmark/studio_benchmark.py [hunk 2]
```diff
@@ -135,17 +140,60 @@ def execute(
...
+            EvaluationLineage[Input, ExpectedOutput, Output, Evaluation]
+        ],
+        trace_ids: list[str],
+    ) -> Sequence[BenchmarkLineage[Input, Output, ExpectedOutput, Evaluation]]:
+        return [
+            self._create_benchmark_lineage(eval_lineage, trace_id)
+            for eval_lineage, trace_id in zip(eval_lineages, trace_ids, strict=True)
+        ]
+
+    def _create_benchmark_lineage(
```
[the remark this answers is not in the document]
Comment (MerlinKallenbornAA) [reply src/intelligence_layer/evaluation/benchmark/studio_benchmark.py:181]: I see the value of it, but and @maxhammeralephalpha tried to make it clean and deemed it not worth the effort atm

File: tests/connectors/studio/test_studio_benchmark.py [hunk 3]
```diff
@@ -138,41 +221,140 @@ def test_get_non_existing_benchmark(studio_client: StudioClient) -> None:
...
+
+def test_submit_benchmark_lineage_uploads_single_lineage(
+    studio_client: StudioClient,
+    with_uploaded_test_trace: str,
+    with_uploaded_benchmark: str,
+    with_uploaded_benchmark_execution: str,
+) -> None:
+    trace_id = with_uploaded_test_trace
+    benchmark_id = with_uploaded_benchmark
+    benchmark_execution_id = with_uploaded_benchmark_execution
```
[the remark this answers is not in the document]
Comment (MerlinKallenbornAA) [reply tests/connectors/studio/test_studio_benchmark.py:246]: Yeah I see your point but it felt awkward to pass them to id parameters with the given names

File: src/intelligence_layer/connectors/studio/studio.py [hunk 4]
```diff
@@ -464,6 +486,78 @@ def create_benchmark_execution(
...
+            data=compressed_data,
+        )
+
+        self._raise_for_status(response)
+        return response.json()
+
+    def _create_post_bechnmark_lineages_request(
+        self, benchmark_lineages: Sequence[BenchmarkLineage]
+    ) -> PostBenchmarkLineagesRequest:
+        return PostBenchmarkLineagesRequest(root=benchmark_lineages)
```
[the remark this answers is not in the document]
Comment (MerlinKallenbornAA) [reply src/intelligence_layer/connectors/studio/studio.py]: Good call, dropped it

File: tests/connectors/studio/test_studio_benchmark.py [hunk 5]
```diff
@@ -138,41 +221,140 @@ def test_get_non_existing_benchmark(studio_client: StudioClient) -> None:
...
+        DummyBenchmarkLineage(
+            trace_id=trace_id,
+            input="input",
+            expected_output="output",
+            example_metadata={"key3": "value3"},
+            output="output",
+            evaluation={"key5": "value5"},
+            run_latency=1,
+            run_tokens=3,
+        ),
```
[the remark this answers is not in the document]
Comment (MerlinKallenbornAA) [reply tests/connectors/studio/test_studio_benchmark.py]: Introduced a function to make the input variable. But good call!

File: tests/connectors/studio/test_studio_benchmark.py [hunk 6]
```diff
@@ -138,41 +221,140 @@ def test_get_non_existing_benchmark(studio_client: StudioClient) -> None:
...
+        benchmark_id=benchmark_id,
+        execution_id=benchmark_execution_id,
+    )
+
+    assert len(lineage_ids.root) == len(lineages)
+    for lineage_id in lineage_ids.root:
+        assert UUID(lineage_id)
+
+
+def test_batch_upload_sends_multiple_requests(
```
[the remark this answers is not in the document]
Comment (MerlinKallenbornAA) [reply tests/connectors/studio/test_studio_benchmark.py]: Fair, removed the test with the single lineage,  improved and renamed this test.

**Review by maxhammeralephalpha — APPROVED** [review]

**Review by NiklasKoehneckeAA — APPROVED** [review]

**Pull request merged**